libraries/resource_provider.rb

```diff
@@ -137,7 +137,7 @@ def remove
     def patch
       invalid_params = @context.operation.nil? || @context.path.nil?
       raise "InvalidParameters: Parameters 'operation' and 'path' must be set for patch" if invalid_params
-      @item.retrieve!
+      return Chef::Log.error("Skipping patch since #{@resource_name} '#{@name}' does not exists") unless @item.retrieve!
```

Comment:
@tmiotto , won't we want it to just fail hard here if it can't be retrieved? I'm imagining it will run into a strange "uri not set" error on the patch if not, right?